app/models/csv_export.rb

```diff
@@ -0,0 +1,39 @@
+class CsvExport < ActiveRecord::Base
+  belongs_to :user
+  STATUS_VALUES = ['pending', 'started', 'finished', 'downloaded', 'failed', 'deleted']
+
+  before_destroy :file_delete
+
+  def status?(state)
+    'ready' == state.to_s ? ['downloaded', 'finished'].include?(status) : status == state.to_s
+  end
+
+  def email
+    user.try :email
+  end
+  
+  def download_name
+    "deploys_#{created_at.strftime "%Y%m%d_%H%M%S"}.csv"
+  end
+
+  def path_file
+    "#{Rails.root}/export/#{id}"
+  end
+
+  def filters
+    filter = JSON.parse(super)
```

Comment:
FYI will blow up on nil or empty ... using the `serializes :filters, JSON` might be easier to read